packages/evolution-frontend/src/services/errorManagement/errorHandling.ts

```diff
@@ -71,32 +71,111 @@ export const handleHttpOtherResponseCode = async (responseCode: number, dispatch
 };
 
 /**
- * Send report of a client side exception to the server
+ * Send messages and exceptions to the server
  *
- * @param interviewId The numeric ID of the current interview
- * @param exception The exception that was thrown
+ * @param clientMessage The message to log or exception that was thrown
+ * @param {Object} options The options object
+ * @param {number|undefined} options.interviewId The numeric ID of the current
+ * interview, if available
+ * @param {'error'|'warn'|'info'|'log'} options.logLevel The log level of the message. default is 'error'
  * @returns
  */
-export const reportClientSideException = async (exception: unknown, interviewId?: number) => {
-    const exceptionMessage =
-        exception instanceof Error
-            ? exception.message
-            : typeof exception === 'object'
-                ? JSON.stringify(exception)
-                : String(exception);
+export const logClientSideMessage = async (
+    clientMessage: unknown,
+    {
+        interviewId = undefined,
+        logLevel = 'error'
+    }: { interviewId?: number; logLevel?: 'error' | 'warn' | 'info' | 'log' } = {}
+) => {
+    const message =
+        clientMessage instanceof Error
```

Comment:
We should really have 2 different paths for Error and for console logging at this but, but I'll let it slip for now. But maybe add a TODO. This started as a way to log exception, which a well structured to simply logging string, so this complexe logic might not be necessary anymore.